### PR TITLE
Latest changes for multitree

### PR DESCRIPTION
**Pull request by cheme**
Latest changes for https://github.com/paritytech/polkadot-sdk/pull/3386

### CODE REVIEW

File: subtrie/src/node.rs [hunk 1]
```diff
@@ -263,17 +270,18 @@ pub enum NodeOwned<H, L> {
 	/// Null trie node; could be an empty root or an empty branch entry.
 	Empty,
 	/// Leaf node; has key slice and value. Value may not be empty.
-	Leaf(NibbleVec, ValueOwned<H, L>),
+	Leaf(NibbleVec, ValueOwned<H, L>, L),
```
Comment (arkpar) [inline subtrie/src/node.rs:273]: What's the reason for storing the location of the node here (and also in `Node`)? I could not find where this is actually used.

File: subtrie/src/triedbmut.rs [hunk 2]
```diff
@@ -69,7 +69,59 @@ fn empty_children<H, L>() -> Box<[Option<NodeHandle<H, L>>; nibble_ops::NIBBLE_L
 	])
 }
 
-pub type TreeRefChangeset<L> = Box<Changeset<TrieHash<L>, <L as TrieLayout>::Location>>;
+/// Alias on changenode over trie layout.
+/// TODO rename (is change node)
+pub type TreeRefChangeset<L> = Option<Changenode<TrieHash<L>, <L as TrieLayout>::Location>>;
+/*
```
Comment (arkpar) [inline subtrie/src/triedbmut.rs]: Please remove commend code

**Comment by arkpar** [comment]
@cheme  Could you take a look at the recent changes as well?

**Comment by arkpar** [comment]
Following this, we should remove Extensions nodes and probably support for `no_std`

File: subtrie/src/node.rs (hunk 1 above)
Comment (cheme) [reply subtrie/src/node.rs:273]: Took me a while to remember, but I think this location is an additional one that is optional on branch and leaf, it holds the location of a child trie root in db (so it does not get overwritten on update (child trie location injection is in substrate iirc, but if the node is touched  then the db will overwrite this n + 1 location so it had to be loaded in node to be set again unchanged).

**Comment by cheme** [comment]
Changes 

> @cheme Could you take a look at the recent changes as well?

looks good

for no_std removal, I guess substrate approach should be followed (remove rstd but and just use alloc or core imports instead), I don't remember but maybe I added this to cumulus, but generally being able to add use this code in runtime sounds relevant to me.

**Comment by arkpar** [comment]
I don't think this crate will ever be used in the runtime. Runtime will continue using the `trie` crate or switch to a binary tree eventually. This crate is specifically adapted to support parity-db and not the runtime. Therefore I think we should simply remove all no_std stuff

**Review by arkpar — APPROVED** [review]

**Pull request merged**